### PR TITLE
📖 Update jobs.md with new k8s v1.27 jobs notes

### PR DESCRIPTION
**Pull request by furkatgofurov7**
Signed-off by: Furkat Gofurov (furkat.gofurov@suse.com)

**What this PR does / why we need it**:
Updates jobs doc according to https://github.com/kubernetes/test-infra/pull/29295

Related to: https://github.com/kubernetes/test-infra/pull/29295
Part of: https://github.com/kubernetes-sigs/cluster-api/issues/8459

/cc @kubernetes-sigs/cluster-api-release-team

### CODE REVIEW

File: docs/book/src/reference/jobs.md [hunk 1]
```diff
@@ -76,7 +76,9 @@ Prow Periodics:
   * GINKGO_FOCUS: `[K8s-Upgrade]`
 * [periodic-cluster-api-e2e-workload-upgrade-1-25-1-26-main] `./scripts/ci-e2e.sh` FROM: `stable-1.25` TO: `stable-1.26`
```
Comment (sbueringer) [inline docs/book/src/reference/jobs.md:77]: Please drop "periodic-cluster-api-e2e-workload-upgrade-1-21-1-22-main" above

(also the link below)

Comment (furkatgofurov7) [reply docs/book/src/reference/jobs.md:75]: Thanks, updated

**Comment by sbueringer** [comment]
/lgtm
/approve

/hold
No strong opinions, but if you want to keep it perfectly in sync you might want to wait until the test-infra PR is merged

**Comment by fabriziopandini** [comment]
/lgtm
Thanks for driving this work @furkatgofurov7!

**Comment by sbueringer** [comment]
/hold cancel

test-infra PR merged, so no reason to hold it anymore

**Comment by furkatgofurov7** [comment]
@sbueringer I think we need to cherry-pick this to release-1.4 branch?

**Comment by sbueringer** [comment]
Yup

/cherry-pick release-1.4

**Pull request merged**